File: app/src/main/java/com/avalon/nsfeo/NSFEOActivity.java
package com.avalon.nsfeo;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.ActionBarActivity;

import com.avalon.nsfeo.fragments.GameDashboardFragment;
import com.avalon.nsfeo.fragments.SessionFragment;
import com.avalon.nsfeo.model.Card;
import com.avalon.nsfeo.model.Deck;
import com.avalon.nsfeo.model.GameSession;
import com.avalon.nsfeo.model.User;
import com.avalon.nsfeo.model.UserSession;
import com.avalon.nsfeo.net.NSFEOService;
import com.avalon.nsfeo.net.NSFEOServiceConnection;
import com.orm.androrm.DatabaseAdapter;
import com.orm.androrm.Model;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class NSFEOActivity extends ActionBarActivity {

	public final static String LOGIN_TRANSACTION_TAG = "NSFEOActivity.this::add R.id.container, SessionFragment";
	public final static String DASHBOARD_TRANSACTION_TAG = "NSFEOActivity.this::add R.id.container, GameDashboardFragment";

	@Override
	protected void onCreate(final Bundle saved_instance) {

		super.onCreate(saved_instance);
		this.setContentView(R.layout.container);

		// Initialize database functionality
		this.initDatabase(
			Card.class,
			UserSession.class,
			User.class,
			Deck.class,
			GameSession.class
		);

		// Initialize (if not available) our trusty service
		final Intent i = new Intent(this, NSFEOService.class);
		final NSFEOServiceConnection connect = new NSFEOServiceConnection();
		this.bindService(i, connect, Context.BIND_AUTO_CREATE);

		final FragmentManager manager = this.getSupportFragmentManager();
		if (!connect.getService().hasLoggedInUser()) {

			final Fragment f = new SessionFragment();
			manager.beginTransaction()
				   .add(R.id.container, f, NSFEOActivity.LOGIN_TRANSACTION_TAG)
				   .commit();
		}
		else {

			final Fragment f = new GameDashboardFragment();
			manager.beginTransaction()
					.add(R.id.container, f, NSFEOActivity.DASHBOARD_TRANSACTION_TAG)
					.commit();
		}

		manager.executePendingTransactions();
	}

	@SafeVarargs
	private final void initDatabase(Class<? extends Model> ... models) {

		DatabaseAdapter.setDatabaseName("NSFEO_DB");
		final DatabaseAdapter adapter = DatabaseAdapter.getInstance(this);

		final List<Class<? extends Model>> models_list = new ArrayList<>();
		Collections.addAll(models_list, models);
		adapter.setModels(models_list);
	}

}

File: app/src/main/kotlin/com/avalon/nsfeo/NSFEOActivity.kt
package com.avalon.nsfeo

import android.os.Bundle
import android.support.v7.app.ActionBarActivity
import com.avalon.nsfeo.fragments.SessionFragment
import com.avalon.nsfeo.model.*
import com.avalon.nsfeo.session.Session
import com.orm.androrm.DatabaseAdapter
import com.orm.androrm.Model
import old.nsfeo.R
import java.util.ArrayList
import java.util.Collections

public class NSFEOActivity: ActionBarActivity() {

	private companion object {
		val LOGIN_TRANSACTION_TAG: String = "NSFEOActivity.this::add R.id.container, LoginFragment"
	}

	override fun onCreate(saved_instance: Bundle?) {

		super.onCreate(saved_instance)
		this.setContentView(R.layout.container)

		// Prepare the environment to connect models to the underlying implementation
		this.initDatabase(
			javaClass<Card>(),
			javaClass<UserSession>(),
			javaClass<User>(),
			javaClass<Deck>(),
			javaClass<GameSession>()
		)

		// Check if there are users, or if a user has been signed in already
		with (this.getSupportFragmentManager()) {

			if (!Session.hasValidSession(this@NSFEOActivity)) {

				val f = SessionFragment()
				this.beginTransaction()
						.add(R.id.container, f, NSFEOActivity.LOGIN_TRANSACTION_TAG)
						.commit()
			}
			else {
				;
			}

			this.executePendingTransactions()
		}
	}

	SafeVarargs
	private fun initDatabase(vararg models: Class<out Model>) {

		DatabaseAdapter.setDatabaseName("NSFEO_DB")
		val adapter = DatabaseAdapter.getInstance(this)

		val models_list = ArrayList<Class<out Model>>()
		Collections.addAll<Class<out Model>>(models_list, *models)
		adapter.setModels(models_list)
	}

}
